dbms/src/TableFunctions/TableFunctionRemote.cpp

```diff
@@ -137,11 +138,38 @@ StoragePtr TableFunctionRemote::executeImpl(const ASTPtr & ast_function, const C
         setIdentifierSpecial(ast);
 
     ClusterPtr cluster;
-    if (!cluster_name.empty())
+    if (!cluster_name.empty() && name != "clusterAllReplicas")
     {
         /// Use an existing cluster from the main config
         cluster = context.getCluster(cluster_name);
     }
+    else if (!cluster_name.empty() && name == "clusterAllReplicas")
+    {
+        std::vector<std::vector<String>> clusterNodes;
+        cluster = context.getCluster(cluster_name);
+        // creating a new topology for clusterAllReplicas
+        const auto & addresses_with_failovers = cluster->getShardsAddresses();
+        const auto & shards_info = cluster->getShardsInfo();
+        auto maybe_secure_port = context.getTCPPortSecure();
+
+        for (size_t shard_index : ext::range(0, shards_info.size()))
+        {
+            const auto & replicas = addresses_with_failovers[shard_index];
+            for (size_t replica_index : ext::range(0, replicas.size()))
+            {
+                std::vector<String> newNode = {replicas[replica_index].host_name};
+                clusterNodes.push_back(newNode);
+            }
+        }
+        cluster = std::make_shared<Cluster>(
+            context.getSettings(),
+            clusterNodes,
```

Comment:
this sounds wrong, it topology each server can have different ports, different usernames, different passwords
also, an integration test would be nice

Reply:
+1. Also stumbled upon this change.

Reply:
@nvartolomei, I understood, But the cluster class is Not accepting different usernames,passwords and port number For each Node. It is taking single username, password and port. correct me if am wrong.